modules/xfeatures2d/src/daisy.cpp

```diff
@@ -0,0 +1,2057 @@
+    // initializes the class: computes gradient and structure-points
+    void initialize();
+
+    void update_selected_cubes();
+
+    int quantize_radius( float rad );
+
+    int filter_size( double sigma );
+
+    // computes scales for every pixel and scales the structure grid so that the
+    // resulting descriptors are scale invariant.  you must set
+    // m_scale_invariant flag to 1 for the program to call this function
+    void compute_scales();
+
+    // Return a number in the range [-0.5, 0.5] that represents the location of
+    // the peak of a parabola passing through the 3 evenly spaced samples.  The
+    // center value is assumed to be greater than or equal to the other values
+    // if positive, or less than if negative.
+    float interpolate_peak( float left, float center, float right );
+
+    // Smooth a histogram by using a [1/3 1/3 1/3] kernel.  Assume the histogram
+    // is connected in a circular buffer.
+    void smooth_histogram( float *hist, int bins );
+
+    // computes pixel orientations and rotates the structure grid so that
+    // resulting descriptors are rotation invariant. If the scales is also
+    // detected, then orientations are computed at the computed scales. you must
+    // set m_rotation_invariant flag to 1 for the program to call this function
+    void compute_orientations();
+
+    // the clipping threshold to use in normalization: values above this value
+    // are clipped to this value for normalize_sift_way() function
+    float m_descriptor_normalization_threshold;
+
+    // computes the sigma's of layers from descriptor parameters if the user did
+    // not sets it. these define the size of the petals of the descriptor.
+    void compute_cube_sigmas();
+
+    // Computes the locations of the unscaled unrotated points where the
+    // histograms are going to be computed according to the given parameters.
+    void compute_grid_points();
+
+    // Computes the locations of the unscaled rotated points where the
+    // histograms are going to be computed according to the given parameters.
+    void compute_oriented_grid_points();
+
+    // smooths each of the layers by a Gaussian having "sigma" standart
+    // deviation.
+    void smooth_layers( float*layers, int h, int w, int layer_number, float sigma );
+
+    // Holds the coordinates (y,x) of the grid points of the region.
+    double** m_grid_points;
+
+    int get_hq() { return m_hist_th_q_no; }
+    int get_thq() { return m_th_q_no; }
+    int get_rq() { return m_rad_q_no; }
+    float get_rad() { return m_rad; }
+
+    // sets the type of the normalization to apply out of {NRM_PARTIAL,
+    // NRM_FULL, NRM_SIFT}. Call before using get_descriptor() if you want to
+    // change the default normalization type.
+    void set_normalization( int nrm_type ) { m_nrm_type = nrm_type; }
+
+    // applies one of the normalizations (partial,full,sift) to the desciptors.
+    void normalize_descriptors( int nrm_type = DAISY::NRM_NONE );
```

Comment:
These are private internals, will never expose such things for public. Did you really want to pass through them Mat &descriptors ? [PENDING]